scripts/validation_utils.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for schema and command line validation"""
+
+import argparse
+import re
+
+
+# For easier development, we allow redefining builtins like
+# --substitutions=PROJECT_ID=foo even though gcloud doesn't.
+KEY_VALUE_REGEX = re.compile(r'^([A-Z_][A-Z0-9_]*)=(.*)$')
+
+
+def get_field_value(container, field_name, field_type):
```

Comment:
During development, I actually ran into type errors multiple times, and every time it was painful:

    entrypoint1: a string, but I can accidentally treat as an sequence
    entrypoint2: [a, list, but, I, can, pass, it, around, for, a, while, before, getting, a, strange, error, because, its, not, a, string]
    feature: off   # This is parsed to the boolean False!
    version1: 3.1 # Parsed to number
    version2: 3.1.1 # Parsed to string

Basically yaml parsing is nuts.